extensions/wikia/ThemeDesigner/ThemeDesignerController.class.php

```diff
@@ -72,35 +66,35 @@ public function executeIndex() {
 		wfProfileOut( __METHOD__ );
 	}
 
-	public function executeThemeTab() {
+	public function themeTab() {
 
 	}
 
-	public function executeCustomizeTab() {
+	public function customizeTab() {
 
 	}
 
-	public function executeWordmarkTab() {
-		$this->faviconUrl = Wikia::getFaviconFullUrl();
+	public function wordmarkTab() {
+		$this->response->setData( [
+			'faviconUrl' => Wikia::getFaviconFullUrl(),
+			'token' => $this->wg->User->getEditToken(),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)